src/test/workerRunner.ts

```diff
@@ -420,7 +410,7 @@ export class WorkerRunner extends EventEmitter {
       await this._runBeforeHooks(test, testInfo);
 
     // Do not run the test when beforeEach hook fails.
-    if (this._isStopped || testInfo.status === 'failed' || testInfo.status === 'skipped')
+    if (testInfo.status === 'failed' || testInfo.status === 'skipped')
```

Comment:
Do we run afterEach hooks when the beforeEach hook fails? I tested out mocha to verify what they do.

```js

before('before', () => {
    console.log('before');
    // throw new Error('before');
});
beforeEach('beforeEach', () => {
    console.log('beforeEach');
    // throw new Error('beforeEach');
});
after('after', () => {
    console.log('after');
    // throw new Error('after');
});
afterEach('afterEach', () => {
    console.log('afterEach');
    // throw new Error('afterEach');
});
it('test', () => {
    console.log('test');
    // throw new Error('test');
});
```

beforeAll throwing causes only afterAll to run
beforeEach throwing causes afterEach/afterAll to run
the test throwing causes everything to run

Reply:
We have `afterEach failure should not prevent other hooks and fixtures teardown` and `beforeEach failure should prevent the test, but not other hooks`. Let me add `beforeAll throwing causes only afterAll to run`.